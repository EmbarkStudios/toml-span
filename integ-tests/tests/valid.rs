use integ_tests::valid;

valid!(comments);
valid!(evil);

mod arrays {
    use super::valid;

    valid!(empty, "thevoid = [[[[[]]]]]");
    valid!(no_spaces, "ints = [1,2,3]");
    valid!(heterogenous, r#"mixed = [[1, 2], ["a", "b"], [1.1, 2.1]]"#);
    valid!(nested, r#"nest = [["a"], ["b"]]"#);
    valid!(ints_and_floats, "ints-and-floats = [1, 1.1]");
    valid!(
        ints_and_arrays,
        r#"arrays-and-ints =  [1, ["Arrays are not integers."]]"#
    );
    valid!(strings_and_ints, r#"strings-and-ints = ["hi", 42]"#);
    valid!(
        one,
        "[[people]]\nfirst_name = \"Bruce\"\nlast_name = \"Springsteen\""
    );
}

mod tables {
    use super::valid;

    valid!(
        implicit_and_explicit_after,
        r"
[a.b.c]
answer = 42

[a]
better = 43
"
    );
    valid!(
        implicit_and_explicit_before,
        r"
[a]
better = 43

[a.b.c]
answer = 42
"
    );
    valid!(implicit_groups, "[a.b.c]\nanswer = 42");
    valid!(implicit_array, "[[albums.songs]]\nname = \"Glory Days\"");
    valid!(
        array_many,
        r#"
[[people]]
first_name = "Bruce"
last_name = "Springsteen"

[[people]]
first_name = "Eric"
last_name = "Clapton"

[[people]]
first_name = "Bob"
last_name = "Seger"
"#
    );
    valid!(
        nested_arrays,
        r#"
[[albums]]
name = "Born to Run"

  [[albums.songs]]
  name = "Jungleland"

  [[albums.songs]]
  name = "Meeting Across the River"

[[albums]]
name = "Born in the USA"
  
  [[albums.songs]]
  name = "Glory Days"

  [[albums.songs]]
  name = "Dancing in the Dark"
"#
    );
    valid!(sub_empty, "[a]\n[a.b]");
    valid!(table_span, "\n[workspace]\nmembers = true\nother = false\n[workspace.foo]\nbar=3\n\n[workspace.bar]\ninline = { foo = 3 }\n\n");
}

mod numbers {
    use super::valid;

    valid!(integers);
    valid!(floats);
}
