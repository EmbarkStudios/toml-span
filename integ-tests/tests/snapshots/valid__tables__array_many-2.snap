---
source: integ-tests/tests/valid.rs
expression: spans
---
note[string]: root_people_0_first_name
  ┌─ array_many:3:15
  │
3 │ first_name = "Bruce"
  │               ^^^^^

note[string]: root_people_0_last_name
  ┌─ array_many:4:14
  │
4 │ last_name = "Springsteen"
  │              ^^^^^^^^^^^

note[table]: root_people_0
  ┌─ array_many:1:1
  │
1 │ 
  │ ^

note[string]: root_people_1_first_name
  ┌─ array_many:7:15
  │
7 │ first_name = "Eric"
  │               ^^^^

note[string]: root_people_1_last_name
  ┌─ array_many:8:14
  │
8 │ last_name = "Clapton"
  │              ^^^^^^^

note[table]: root_people_1
  ┌─ array_many:1:1
  │
1 │ 
  │ ^

note[string]: root_people_2_first_name
   ┌─ array_many:11:15
   │
11 │ first_name = "Bob"
   │               ^^^

note[string]: root_people_2_last_name
   ┌─ array_many:12:14
   │
12 │ last_name = "Seger"
   │              ^^^^^

note[table]: root_people_2
  ┌─ array_many:1:1
  │
1 │ 
  │ ^

note[array]: root_people
  ┌─ array_many:1:1
  │
1 │ 
  │ ^

note[table]: root
   ┌─ array_many:1:1
   │  
 1 │ ╭ 
 2 │ │ [[people]]
 3 │ │ first_name = "Bruce"
 4 │ │ last_name = "Springsteen"
   · │
12 │ │ last_name = "Seger"
13 │ │ 
   │ ╰^


